pkg/config/types/v1alpha1/cluster_config.go

```diff
@@ -9,14 +9,15 @@ import (
 
 	"github.com/talos-systems/talos/pkg/config/cluster"
 	"github.com/talos-systems/talos/pkg/config/machine"
+	"github.com/talos-systems/talos/pkg/constants"
 	"github.com/talos-systems/talos/pkg/crypto/x509"
 )
 
 // ClusterConfig reperesents the cluster-wide config values
 type ClusterConfig struct {
 	ControlPlane                  *ControlPlaneConfig               `yaml:"controlPlane"`
 	ClusterName                   string                            `yaml:"clusterName,omitempty"`
-	Network                       *ClusterNetworkConfig             `yaml:"network,omitempty"`
+	ClusterNetwork                *ClusterNetworkConfig             `yaml:"network,omitempty"`
```

Comment:
I don’t like it but it’s required to be something other than `Network` since `Network` is a required method to satisfy the `Configurator` interface.

Reply:
Ah, this would be where we start to see the pure interface assertions break down, I'd guess.  If we used a two-step process with a metadata interface (like kubernetes), we could have a type/kind field.  That's beyond the scope of this PR, though, and this really was just a nit.

Reply:
Before 0.3 I’d like to get something better though. At least naming conventions